components/services/bat_ads/bat_ads_client_mojo_bridge.cc

```diff
@@ -290,59 +290,84 @@ void BatAdsClientMojoBridge::Log(const char* file,
 }
 
 absl::optional<base::Value> BatAdsClientMojoBridge::GetProfilePref(
-    const std::string& path) const {
+    const std::string& path) {
   if (!bat_ads_client_associated_receiver_.is_bound()) {
-    return absl::nullopt;
+    return GetCachedProfilePrefValue(path);
   }
 
   absl::optional<base::Value> value;
-  bat_ads_client_associated_receiver_->GetProfilePref(path, &value);
+  if (!bat_ads_client_associated_receiver_->GetProfilePref(path, &value)) {
+    return GetCachedProfilePrefValue(path);
+  }
+
+  if (value) {
```

Comment:
would the above condition not fail if value is null?

Reply:
yes, it is possible based on our Mojom interface definition:

```
GetProfilePref(string path) => (mojo_base.mojom.Value? value);
```